src/database/DatabaseManager.js

```diff
@@ -344,60 +421,183 @@ export class DataBaseManager {
+     */
+    async saveQuestionAdmin(aQuestion, aIdDiscussion, aUsername){
+        let saved = await this.saveQuestion(aQuestion,aIdDiscussion);
+        if(!saved){
+            return false;
+        }
+        // Save the question as an admin Question in the database
+        let idAdmin = await this.getAdminId(aUsername);
+        if(idAdmin == null){
+            logger.debug(`Error when looking for username id`);
+            return false;
+        }
+
+        const questionAdminSave = new QuestionAdmin({
+            id: {
+                refQuestion: aQuestion.id,
+                refDiscussion: aIdDiscussion
+            },
+            administrator: idAdmin
+        });
+
+        await questionAdminSave.save()
+            .then(questionAdminSave => logger.debug(`Question of admin saved ${questionAdminSave}`))
+            .catch(err => {
+                logger.debug(`Error when saving Question of admin id = ${aQuestion.id}`);
+                console.log(err);
+                saved = false;
+            });
+        // If the save went wrong we exit the function and return false
+        if(!saved){
+            return false;
+        }
+        return saved;
+    }
+
+    /**
+     * Save a question suggested by a user in the database
+     * @param aQuestion object Question that represents the Question to save
+     * @param aIdDiscussion integer that is the id of the Discussion related to the question
+     * @returns {Promise<boolean>} true if the save went well false otherwise
+     */
+    async saveQuestionUser(aQuestion, aIdDiscussion){
+        // Save the question as a Question
+        let saved = await this.saveQuestion(aQuestion,aIdDiscussion);
+        // If the save of the question doesn't went well error
+        if(!saved){
+            return false;
+        }
+        const questionSuggestionSave = new QuestionSuggestion({
+            id: {
+                refQuestion: aQuestion.id,
+                refDiscussion: aIdDiscussion
+            }
+        });
+
+        // Save the question as a QuestionSuggestion
+        await questionSuggestionSave.save()
+            .then(questionAdminSave => logger.debug(`Question of user saved ${questionAdminSave}`))
+            .catch(err => {
+                logger.debug(`Error when saving Question of user id = ${aQuestion.id}`);
+                console.log(err);
+                saved = false;
+            });
+        // If the save went wrong we exit the function and return false;
+        if(!saved){
+            return false;
+        }
+        return saved;
+    }
+
+    /**
+     * Update a question as approved in the database
+     * @param aQuestionId integer that is the id of the question that has been approved
+     * @param aDiscussionId integer that is the id of teh discussion related to the question
+     * @returns {Promise<boolean>} true if the update went well false otherwise
+     */
+    async saveApprovedQuestion(aQuestionId, aDiscussionId){
```

Comment:
I feel like this function name is misleading. I would suggest to change it to `approveQuestion(...)`. What do you think ?